### PR TITLE
[Structural] clean response fct tests

### PR DESCRIPTION
**Pull request by philbucher**
For the transition of using the solvers from the EigenSolversApp instead of the one from the ExternalSolversApp I noticed that the Responsefct tests have the superLU hardcoded.
Can you please change this (probably to the sparse-LU) and add the corresponding skiptests?

I also started some cleanup, can you please finish it?
BTW `controlledExecutionScope` should be used from `KratosUnittest`

Thx

### CODE REVIEW

**Review by armingeiser — DISMISSED** [review]
Thanks!

File: applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_tests/adjoint_beam_structure_3d2n/beam_test_local_stress_adjoint_parameters.json [hunk 1]
```diff
@@ -36,9 +36,6 @@
             },
         "echo_level"                   : 0,
         "rotation_dofs"                      : true,
-        "linear_solver_settings"       : {
-            "solver_type"         : "ExternalSolversApplication.super_lu"
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_tests/adjoint_beam_structure_3d2n/beam_test_local_stress_adjoint_parameters.json:40]: I would still prescribe this like I did this in the `structural_mechanics_test_factory` I suggest to briefly discuss in person.

**Review by philbucher — COMMENTED** [review]
nice, thx 
One minor thing to be discussed in person

File: applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_test_factory.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+            # solve primal problem
+            model_primal = KratosMultiphysics.Model()
+            primal_analysis = structural_mechanics_analysis.StructuralMechanicsAnalysis(model_primal, self.primal_parameters)
+            primal_analysis.Run()
+            # create adjoint analysis
+            model_adjoint = KratosMultiphysics.Model()
+            self.adjoint_analysis = structural_mechanics_analysis.StructuralMechanicsAnalysis(model_adjoint, self.adjoint_parameters)
+            self.adjoint_analysis.Initialize()
+
+    def verify_linear_solver(self, parameter_file):
```
Comment (armingeiser) [inline applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_test_factory.py]: Minor: is it possible to reuse this function from the SturcutMechFactory somehow? Maybe also slightly modifying the implementation there?

File: applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_test_factory.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+            kratos_utilities.DeleteFileIfExisting(self.model_part_name + "-1.1000.h5")
+
+
+@KratosUnittest.skipUnless(has_hdf5_application,"Missing required application: HDF5Application")
+class TestAdjointSensitivityAnalysisBeamStructureLocalStress(AdjointSensitivityAnalysisTestFactory):
+    primal_file_name = "adjoint_sensitivity_analysis_tests/adjoint_beam_structure_3d2n/beam_test_parameters.json"
+    adjoint_file_name = "adjoint_sensitivity_analysis_tests/adjoint_beam_structure_3d2n/beam_test_local_stress_adjoint_parameters.json"
+
+    def test_execution(self):
+        self.solve_adjoint_problem()
```
Comment (armingeiser) [inline applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_test_factory.py]: This seems a little strange, couldn't you do that in setup as well? What do you think?

**Review by armingeiser — COMMENTED** [review]
Looks good! Thanks a lot, only some minor comments

File: applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_test_factory.py (hunk 2 above)
Comment (MFusseder) [reply applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_test_factory.py]: good hint. I modified it.

File: applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_test_factory.py (hunk 3 above)
Comment (MFusseder) [reply applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_test_factory.py]: good hint. I modified it.

**Review by armingeiser — APPROVED** [review]

**Pull request merged**